components/automate-ui/src/app/entities/download-reports/download-reports.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../../environments/environment';
+
+const REPORT_LIST_API_URL = environment.download_report_list_url;
+
+@Injectable({ providedIn: 'root'})
+export class DownloadReportsService {
+  showOpenReport = false;
+  reportList = [];
+  private units = ['bytes', 'kb', 'mb', 'gb'];
+  constructor(private httpClient: HttpClient) {}
+
+  onReportOpenClick() {
+    this.handleReportList();
+    this.showOpenReport = true;
+  }
+
+  onReportCloseClick() {
+    this.showOpenReport = false;
+  }
+
+  handleReportList() {
+    this.fetchReportList().subscribe((responseData) => {
+      if (responseData && responseData['data'] && responseData['data'].length > 0) {
+        this.reportList = responseData['data'];
+        this.checkReportStatus();
+      } else {
+        this.reportList = [];
+      }
+    });
+  }
+
+  fetchReportList() {
+    const url = `${REPORT_LIST_API_URL}/requests`;
```

Comment:
what is the reason behind defining the  url as 
download_report_list_url: '/api/v0/reportmanager',
and appending '/requests' here. 
why can't we define as below
download_report_list_url: '/api/v0/reportmanager/requests',

Reply:
As we discussed, we are following common pattern in UI side. Created environment variable for this URL "'/api/v0/reportmanager" and '/requests' is used only in service file accordingly. In future we can reuse this variable with different endpoints.